server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -340,7 +340,7 @@ public Tablet(final TabletServer tabletServer, final KeyExtent extent,
 
     tabletMemory = new TabletMemory(this);
 
-    if (!logEntries.isEmpty()) {
+    if (!logEntries.isEmpty() && !isBeingDeleted()) {
```

Comment:
It's not clear without reading a bunch of code what thy this block is being skipped under these circumstances. A comment would help. Is this change just to avoid attempting WAL recovery on a table being deleted, if such a tablet is being loaded?

Reply:
I can add a comment. 

> Is this change just to avoid attempting WAL recovery on a table being deleted, if such a tablet is being loaded?

Exactly. This is one optimization of least resistance for deletes during a tablet load. I really didn't want to try and short circuit the Tablet loading and risk breaking something.